doc/api/n-api.md

```diff
@@ -0,0 +1,3151 @@
+# N-API
+
+> Stability: 1 - Experimental
+
+
+N-API is a newer API for building native Addons.  It is independent from
+the underlying JavaScript runtime (ex v8) and is maintained as part of
+Node.js itself.  This API will be Application Binary Interface (ABI) stable
+across versions of Node.js.  It is intended to insulate Addons from
+changes in the underlying JavaScript engine and allow modules
+compiled for one version to run on later versions of Node.js without
+recompilation.
+
+Addons are built/packaged with the same approach/tools
+outlined the second titled  [C/C++ Addons](addons.html).
+The only difference is the set of APIs that are used by the native code.
+Instead of using the V8 or [Native Abstractions for Node.js][] APIs,
+the functions available in the N-API are used instead.
+
+
+APIs exposed by the N-API surface are generally used to create and manipulate
+JavaScript values. Concepts and operations generally map to ideas specified 
+in the ECMA262 Language Specification. The APIs have the following 
+properties:
+- All N-API calls return a status code of type `napi_status`. This
+  status can be used to determine whether the API call succeeded or failed.
+- The actual value being returned by the API is returned in an out parameter.
+- All JavaScript values are abstracted behind an opaque type named 
+  `napi_value`.
+- In case of an error status code, additional information can be obtained 
+  using `napi_get_last_error_info`. More information can be found in the error
+  handling section [Error Handling][Error Handling].
+
+The documentation for N-API is structured as follows:
+
+* [Basic N-API Data Types][]
+* [Error Handling][]
+* [Object Lifetime Management][]
+* [Module Registration][]
+* [Working with JavaScript Values][]
+* [Working with JavaScript Values - Abstract Operations][]
+* [Working with JavaScript Properties][]
+* [Working with JavaScript Functions][]
+* [Object Wrap][]
+* [Aynchronous Operations][]
+
+The N-API is a C based API in order to allow us to ensure ABI stability
+across Node.js versions and different compiler levels.  However, we also
+understand that a C++ API can be easier/faster to use in many
+cases.  To support these cases we expect there to be one or more
+C++ wrapper modules that provide an in-lineable C++ API that results in
```

Comment:
This whole sentence is kind of confusing. Consider breaking into multiple sentences?